x-pack/platform/plugins/shared/fleet/server/services/agents/crud.ts

```diff
@@ -491,23 +491,28 @@ export const getByIds = async (
   const agentsHits = await getAgentsById(esClient, soClient, agentIds);
   const currentNamespace = getCurrentNamespace(soClient);
   const response: Agent[] = [];
+  const throwNotFoundError = (agentId: string): never => {
+    throw new AgentNotFoundError(`Agent ${agentId} not found`, { agentId });
+  };
 
   for (const agentHit of agentsHits) {
-    let throwError = false;
-
-    if ('notFound' in agentHit && !options?.ignoreMissing) {
-      throwError = true;
-    } else if ((await isAgentInNamespace(agentHit as Agent, currentNamespace)) !== true) {
-      throwError = true;
+    const wasFound = !('notFound' in agentHit);
+    if (!wasFound) {
+      if (!options?.ignoreMissing) {
+        throwNotFoundError(agentHit.id);
+      }
+      continue;
     }
 
-    if (throwError) {
-      throw new AgentNotFoundError(`Agent ${agentHit.id} not found`, { agentId: agentHit.id });
+    const isAccessible = await isAgentInNamespace(agentHit as Agent, currentNamespace);
+    if (!isAccessible) {
+      if (!options?.ignoreMissing) {
+        throwNotFoundError(agentHit.id);
+      }
+      continue;
     }
```

Comment:
nit: I liked the previous version better. It was easier to read and understand. I actually find this kind of if logic confusing. The following 

```json5
if (a) { 
 if(b) {
  // do something
 }
 // do else
}
 ```

is basically same as
```
if (a && b) {
  // do something
} else if (a) {
  // do else
}
``` 
which would be easier to reason with. Either way I thought the previous version was better in terms of readability and with less redundant code.

cc @tomsonpl @juliaElastic